### PR TITLE
Dropdown trigger size

### PR DESCRIPTION
**Pull request by philschanely**
## Description

Adds ability to set a custom width for dropdown triggers. Particularly useful for dropdowns in toolbars that otherwise adjust to the size of the selected item.

## Screenshots

Screen Shot 2022-03-25 at 4 02 21 PM

## Testing in `sage-lib`
See [Components > Dropdown](http://localhost:4000/pages/component/dropdown) and [Storybook](http://localhost:4100/?path=/story/sage-dropdown--default)

## Testing in `kajabi-products`

1. (LOW) Adds ability to set custom width on Dropdown triggers

## Related

https://kajabi.atlassian.net/browse/SAGE-235

### CODE REVIEW

File: docs/lib/sage_rails/app/views/sage_components/_sage_dropdown.html.erb [hunk 1]
```diff
@@ -1,6 +1,5 @@
 <%
-trigger_type = component.trigger_type.present? ? component.trigger_type : nil
-trigger_configs = component.trigger.present? ? component.trigger : { type: trigger_type }
```
Comment (philschanely) [inline docs/lib/sage_rails/app/views/sage_components/_sage_dropdown.html.erb:3]: These lines were problematic for legacy setup in `kp` so I cleaned it up and sync with change on line 38->33

File: docs/lib/sage_rails/app/views/sage_components/_sage_dropdown.html.erb [hunk 2]
```diff
@@ -20,13 +19,9 @@ trigger_configs = component.trigger.present? ? component.trigger : { type: trigg
 >
   <div aria-hidden="true" class="sage-dropdown__screen"></div>
 
-  <% if component.trigger.present? %>
```
Comment (philschanely) [inline docs/lib/sage_rails/app/views/sage_components/_sage_dropdown.html.erb:23]: Simplified logic here for common outcome.

File: docs/lib/sage_rails/app/views/sage_components/_sage_dropdown_trigger.html.erb [hunk 3]
```diff
@@ -3,13 +3,15 @@
...
     <%= "sage-dropdown__trigger--#{component.type}" if component.type.present? %>
+    <%= "sage-dropdown__trigger--custom-width" if component.width.present? %>
     <%= component.generated_css_classes %>
   "
+  <% if component.width.present? %>
+    style="--sage-dropdown-trigger-width: <%= component.width %>"
+  <% end %>
   <%= component.generated_html_attributes %>
 >
-  <% if component.content.present? %>
```
Comment (philschanely) [inline docs/lib/sage_rails/app/views/sage_components/_sage_dropdown_trigger.html.erb:10]: Switch logic here to improve legacy use cases in sync with related adjustments in `sage_dropdown.html.erb`

File: packages/sage-assets/lib/stylesheets/components/_button.scss [hunk 4]
```diff
@@ -320,6 +323,20 @@ $-btn-loading-min-height: rem(36px);
     }
   }
 
+  .sage-toolbar__group > .sage-dropdown .sage-dropdown__trigger > & {
```
Comment (philschanely) [inline packages/sage-assets/lib/stylesheets/components/_button.scss:326]: Caught a further tweak needed to get the styles right within Toolbar group context.

**Review by monicawheeler — APPROVED** [review]
LGTM

**Pull request merged**